src/web/lib/braze/BrazeMessages.ts

```diff
@@ -1,48 +1,88 @@
-type Callback = (message: Message) => void;
+/* eslint-disable max-classes-per-file */
+
+import type appboy from '@braze/web-sdk-core';
 
 type Extras = Record<string, string>;
 
-interface Appboy {
-	subscribeToInAppMessage: (callback: Callback) => string;
+interface BrazeMessagesInterface {
+	getMessageForBanner: () => Promise<BrazeMessage>;
+	getMessageForEndOfArticle: () => Promise<BrazeMessage>;
 }
 
-interface Message {
-	extras?: Extras;
-	triggerId?: string; // This keeps Typescript happy, because Message could be an InAppMessage or a ControlMessage
+class BrazeMessage {
+	appboy: typeof appboy;
+
+	message: appboy.InAppMessage;
+
+	constructor(message: appboy.InAppMessage, ab: typeof appboy) {
+		this.message = message;
+		this.appboy = ab;
```

Comment:
I don't think there's an actual crossover but it might safer to not use the 'ab' name just because it denotes ab testing in other parts of the codebase. Perhaps later if an ab test were to be added here it could be tricky.